### PR TITLE
Fix: KafkaSink ContentMode not checking 

### LINKED ISSUE #2951: KafkaSinks ContentMode ignored
**Describe the bug**

The default content mode when installing a Kafka Sink is `binary`:

https://github.com/knative-sandbox/eventing-kafka-broker/blob/9459144e95bd13df164340970a96b58a2761daa4/control-plane/pkg/apis/eventing/v1alpha1/kafka_sink_defaults.go#L28-L32

In #2932 we migrated an e2e test for the Sink, with a job for publishing to the Sink using `structured` content mode, without changing the default in Sink before. This test still succeeds, so 

* No verification is happening, or
* There is a bug in the sink implementation for changing content mode

 **To Reproduce**
Steps to reproduce the behavior.

Run the `TestBrokerTirggersSink` test from #2932

 **Knative release version**
On main branch (today with 5475bd8641cb0b732dca4ef1adc2c67d0d68e358)

### PR DESCRIPTION
**Pull request by debasishbsws**
Fixes #2951

## Proposed Changes

- set the sink content mode to `structured`
- change the `Assert` to `IsDone()`, so that, the test fails if the job is uncompleted.
- change the consumer so it gets into the complete state after receiving events from the 2 triggers.
- change `delete_cm_test.go` and `kafka_sink_test.go` where previously both excepting structured format by default(should be binary).
   
  
this is a replacement of PR #2998

### CODE REVIEW

**Review by creydr — COMMENTED** [review]
Hello @debasishbsws,
thanks for your contributions. Can we have an additional e2e test just to make sure the content modes of sink and producer are aligned (to avoid this in future)? That way we should ensure the following (from #2951) is fixed:
> - No verification is happening, or
> - There is a bug in the sink implementation for changing content mode

File: test/test_images/kafka-consumer/consumer.go [hunk 1]
```diff
@@ -76,15 +76,24 @@ func main() {
...
-				exitError <- err
+			select {
+			case <-ctx.Done():
+				log.Println("closing consumer")
+				isConsumerClose <- true
 				return
+			default:
+				err := consumer.Consume(ctx, []string{envConfig.Topic}, handler)
+				if err == sarama.ErrClosedConsumerGroup {
+					return
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply test/test_images/kafka-consumer/consumer.go:91]: An `ErrClosedConsumerGroup` error comes when the consumer is closed. So I thought there might be a possibility that this code may run before the consumer is ready to consume any event and we may get this error. And as this condition is already there than I thought just not to touch it. But it seems that adding this line to the code is also fine.

File: test/test_images/kafka-consumer/consumer.go [hunk 2]
```diff
@@ -110,10 +119,24 @@ func main() {
...
+L:
+	for {
+		log.Println("Waiting to close ConsumerGroup")
+		select {
+		case ent := <-events:
+			log.Println("Received event", ent.ID())
+		case <-isConsumerClose:
+			break L
+		}
+	}
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply test/test_images/kafka-consumer/consumer.go:139]: yes we could do that.

File: test/test_images/kafka-consumer/consumer.go [hunk 3]
```diff
@@ -76,15 +76,24 @@ func main() {
 
 	exitError := make(chan error)
 
+	isConsumerClose := make(chan bool)
```
[the remark this answers is not in the document]
Comment (debasishbsws) [reply test/test_images/kafka-consumer/consumer.go]: done.

**Comment by debasishbsws** [comment]
Hey @creydr, I am not quite sure what you mean by "content modes of sink and producer are aligned" Isn't the content mode responsible for the output event type from the sink?

**Comment by creydr** [comment]
Hey @debasishbsws ,
I meant a test for the following scenario: we have a KafkaSink with content-mode X and a producer (e.g. a Job) with content-mode Y. In case the producer sends an event to the KafkaSink, I would expect it to fail, as the content-mode of the sent-event and the content-mode what the KafkaSink expects differ.
This is kind of what happened in #2951. So it would be great to have a test for it to ensure the producer-job fails, when it sends and event of content-mode Y to a KafkaSink with content-mode X.

This should then also verify that your PR is correct.

Does this make it clearer?

**Comment by matzew** [comment]
Should we rebase this, @debasishbsws ?

**Comment by debasishbsws** [comment]
Hey, It has been almost a year, I am not working on this so It might be better if someone else takes it from here. So I am closing this PR.

**Pull request closed without merging**